samcli/hook_packages/terraform/hooks/prepare.py

```diff
@@ -276,9 +278,66 @@ def _generate_custom_makefile(
     cfn_resources: dict
         CloudFormation resources
     output_directory_path: str
-        the output directory path to write the generated metadata and makefile
+        the output directory path to write the generated makefile
+    terraform_application_dir: str
+        the terraform project root directory
+    """
+
+    # create output directory if it doesn't exist
+    if not os.path.exists(output_directory_path):
+        os.makedirs(output_directory_path, exist_ok=True)
+
+    # copy copy_terraform_built_artifacts.py script into output directory
+    shutil.copy("../copy_terraform_built_artifacts", output_directory_path)
+
+    # check python is installed and get python command
+    python_command_name = _get_python_command_name()
+
+    # create makefile
+    makefilePath = os.path.join(output_directory_path, "Makefile")
+    with open(makefilePath, "a+") as makefile:
+        for sam_metadata_resource in sam_metadata_resources:
+            _write_makefile_rule_for_lambda_resource(
```

Comment:
how will you determine the Lambda resource that you are going to generate a building rule for it?

Reply:
planning to use similar logic of the enrich function, using `_get_relevant_cfn_resource`

Reply:
My issue is repeating this logic, so in case if we have to add more complex logic to determine the resource, we will repeat this logic, and may cause performance issues. 

Can we move the makefile rules generation to the enrichment method, or change the enrichment method to add the found resource to each sam_metadata object

Reply:
it doesn't seem very complex or performance heavy right now, though 🤔 . Can we just leave as is, and if at some point in the future we introduce changes that will impact performance then we just refactor at that point?

Reply:
or we can also do it as a followup PR

Reply:
I prefer to do that now better than later. The effort to do the change later will be more expensive than now. I agree that the performance now is not heavy, but if we changed how we determine the cfn resource later it may be heavy specially that we already defined this information before, and this fact will never change.

Reply:
done

Reply:
thanks